core/sql/executor/ExHdfsScan.cpp

```diff
@@ -1700,6 +1714,138 @@ char * ExHdfsScanTcb::extractAndTransformAsciiSourceToSqlRow(int &err,
   return NULL;
 }
 
+void ExHdfsScanTcb::computeRangesAtRuntime()
```

Comment:
I suppose this computation is being repeated by each esp. Is there a way by which it can done once and shared (say computed during fixup)? If possible, this would be a minor enhancement for later. I do see some comment indicating that only portions relevant to this esp are computed. Maybe some comment including a statement on which locality_balance_level this corresponds to will help.